Classes/Mvc/Controller/CommandController.php

```diff
@@ -259,10 +259,10 @@ protected function callCommandMethod()
             $preparedArguments[] = $argument->getValue();
         }
         $commandResult = $this->{$this->commandMethodName}(...$preparedArguments);
-        if (is_string($commandResult) && $commandResult !== '') {
-            $this->response->appendContent($commandResult);
-        } elseif (is_object($commandResult) && method_exists($commandResult, '__toString')) {
-            $this->response->appendContent((string)$commandResult);
+        if ($commandResult !== null) {
+            $this->outputLine((string)$commandResult);
+            $this->outputLine('<warning>Returning a string from a command method is deprecated.</warning>');
```

Comment:
This not a good solution since it changes the output of commands. If you e.g. use the output of a command for another command, this would be broken. This should use the regular deprecation logic of TYPO3 instead.

Reply:
OK, now I'm lost. Can you create a ticket with a use case you see failing? Or re-open the existing ticket?

Reply:
Simple usecase:

    curl -sS bin/typo3cms foo:geturl

Whereas `foo:geturl` normally only returns an URL, it now returns additional lines with a warning and thus breaks the web request.